### PR TITLE
Implement Reset Password Feature

### PR DESCRIPTION
**Pull request by kkumar-gcc**
This pull request is in reference to issue #4 - Reset password.

I have implemented a new "forgot password" view where users can enter their email address and request a password reset. Upon submission, an email will be sent to the user's email address containing a password reset link.

I have tested the feature thoroughly and it is working as expected.

Please review the changes and let me know if any modifications are needed.

Thank you!

### CODE REVIEW

**Comment by devkiran** [comment]
@kkumar-gcc thanks for the PR! We'll review it as soon as we can. 👍🏻

**Comment by devkiran** [comment]
@kkumar-gcc Can you run prettier on the code changes and fix the conflicts?

**Comment by kkumar-gcc** [comment]
@devkiran I have run prettier and fixed the conflicts.

File: components/interfaces/Auth/resetPasswordForm.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        'Passwords must match'
+      ),
+    }),
+    onSubmit: async (values) => {
+      setSubmitting(true);
+
+      const { password } = values;
+
+      try {
+        var response = await fetch('/api/auth/reset-password', {
```
Comment (devkiran) [inline components/interfaces/Auth/resetPasswordForm.tsx:34]: Use `const` instead of `var` to keep the code consistent.

File: package.json [hunk 2]
```diff
@@ -32,6 +32,8 @@
     "bcryptjs": "2.4.3",
     "classnames": "2.3.2",
     "cookies-next": "2.1.1",
+    "crypto-js": "^4.1.1",
+    "daisyui": "^2.51.6",
```
Comment (devkiran) [inline package.json:36]: What is the need of daisyui here?

File: package.json [hunk 3]
```diff
@@ -32,6 +32,8 @@
     "bcryptjs": "2.4.3",
     "classnames": "2.3.2",
     "cookies-next": "2.1.1",
+    "crypto-js": "^4.1.1",
```
Comment (devkiran) [inline package.json:35]: Please lock the package version/

File: package.json [hunk 4]
```diff
@@ -60,6 +62,7 @@
     "@testing-library/react": "13.4.0",
     "@trivago/prettier-plugin-sort-imports": "^4.0.0",
     "@types/bcryptjs": "2.4.2",
+    "@types/crypto-js": "^4.1.1",
```
Comment (devkiran) [inline package.json:65]: Make sure you lock the package versions.

**Review by devkiran — CHANGES_REQUESTED** [review]
Can you please take a look the following comments? I have added a few more to the code.

- Missing the link to `/auth/forgot-password` from the login page. Keep it under the login box if possible.
- I think you forgot to the translation texts to `locale/en/common.json` for new keys you added. Can you please add them?
- I noticed the password reset token is expiring before 60 minutes. Can you please check? Looks like a bug.
- Lock the package versions in `package.json` to avoid any breaking changes in the future.

File: package.json (hunk 4 above)
Comment (kkumar-gcc) [reply package.json:65]: I have locked package version.

File: package.json (hunk 2 above)
Comment (kkumar-gcc) [reply package.json]: It is being used in `tailwind.config.js`

**Review by devkiran — APPROVED** [review]

**Pull request merged**